articles/tokens/id-token.md

```diff
@@ -0,0 +1,107 @@
+
+## How to validate an ID token
+
+The way to validate an ID token depends on the hash algorithm used by your Client:
+- If you used `HS256` then the token is signed with the Client Secret, using the HMAC algorithm. You can verify the signature using the Client Secret value, which you can find at the _[Client Settings](${manage_url}/#/clients/${account.clientId}/settings)_ page.
+- If you used `RS256` then the token is signed with a public/private key pair, using RSA. You can verify the signature using the Public Key or Certificate, which you can find at the _[Client Settings](${manage_url}/#/clients/${account.clientId}/settings) > Show Advanced Settings > Certificates_ page.
+
+To check or update the algorithm your Client uses go to _[Client Settings](${manage_url}/#/clients/${account.clientId}/settings) > Show Advanced Settings > OAuth > JsonWebToken Signature Algorithm_. The most secure practice, and our recommendation, is to use `RS256`.
+
+## How to control the contents of an ID token
+
+In order to retrieve an `id_token` the `responseType` should be set to `id_token`, both for client-side and server-side authentication flows.
+
+::: panel-warning OAuth2 as a Service
+If you are not using *OAuth2 as a Service*, then you should use `code` for server side flows and `token` for client side flows.
+:::
+
+The attributes included in the issued `id_token` are controlled by the use of a [parameter called `scope`](/scopes).
+- If `scope` is set to `openid`, then the `id_token` will contain only the `iss`, `sub`, `aud`, `exp` and `iat` claims.
+- If `scope` is set to `openid email`, then the `id_token` will contain additionally the `email` and `email_verified` claims.
+- If `scope` is set to `openid profile`, then the `id_token` will contain all default profile Claims, which are: `name`, `family_name`, `given_name`, `middle_name`, `nickname`, `preferred_username`, `profile`, `picture`, `website`, `gender`, `birthdate`, `zoneinfo`, `locale`, and `updated_at`.
+
+If you are using Lock, the `options` object used in Lock’s instantiation can specify optional [authentication parameters](/libraries/lock/v10/customization#auth-object-) as follows:
+
+```js
+var options = {
+  auth: {
+    responseType: 'id_token',
+    params: {scope: 'openid email'}
+  }
+};
+
+var lock = new Auth0Lock(
+  ${account.clientId},
+  ${account.namespace},
+  options
+);
+
+lock.show();
+```
+
+The `id_token` will contain only the claims specified as the value of the `scope` parameter.
+
+**NOTE:** You can also add claims to the `id_token` using [Rules](/rules), with the following format: `context.idToken['http://my-custom/claim'] = 'some-value'`.
```

Comment:
Replaced note with panel and added link to [OAuth 2.0 API Authorization](https://auth0.com/docs/api-auth/tutorials/configuring-tenant-for-api-auth).